sky/clouds/service_catalog/gcp_catalog.py

```diff
@@ -385,34 +385,41 @@ def list_accelerators(
             new_results[acc_name] = acc_info
     results = new_results
 
-    # Unlike other GPUs that can be attached to different sizes of N1 VMs,
-    # A100 GPUs can only be attached to fixed-size A2 VMs,
-    # and L4 GPUs can only be attached to G2 VMs.
-    # Thus, we can show their exact cost including the host VM prices.
-
-    acc_infos: List[common.InstanceTypeInfo] = sum(
-        [results.get(a, []) for a in _ACC_INSTANCE_TYPE_DICTS], [])
-    if not acc_infos:
-        return results
+    # If we can figure out which instances are to be used then we
+    # can attach them here.
+    acc_infos: List[common.InstanceTypeInfo] = sum(results.values(), [])
 
     new_infos = defaultdict(list)
     for info in acc_infos:
         assert pd.isna(info.instance_type) and pd.isna(info.memory), acc_infos
-        vm_types = _ACC_INSTANCE_TYPE_DICTS[info.accelerator_name][
-            info.accelerator_count]
+        if info.accelerator_name.startswith('tpu'):
+            new_infos[info.accelerator_name].append(
+                info._replace(instance_type='TPU-VM',))
+            continue
+        vm_types, _ = get_instance_type_for_accelerator(info.accelerator_name,
+                                                        info.accelerator_count,
+                                                        region=region_filter,
+                                                        use_spot=True)
+        # The acc name & count in `info` are retrieved from the table so we
+        # could definitely find a column in the original table
+        # Additionally the way get_instance_type_for_accelerator works
+        # we will always get either a specialized instance type (L274 - L284)
+        # or a default instance type (L300). So we can safely assume that
```

Comment:
```suggestion
        # we will always get either a specialized instance type
        # or a default instance type. So we can safely assume that
```
We should avoid using absolute reference since this file might be changed later; probable we could just remove them.